benchmarks/sindex.py

```diff
@@ -0,0 +1,44 @@
+import random
+
+from geopandas import GeoDataFrame, GeoSeries
+from shapely.geometry import Point, Polygon
+import numpy as np
+
+
+class Bench:
+    def setup(self, *args):
+        triangles = GeoSeries(
+            [
+                Polygon([(random.random(), random.random()) for _ in range(3)])
+                for _ in range(100)
+            ]
+        )
+
+        points = GeoSeries(
+            [
+                Point(x, y)
+                for x, y in zip(np.random.random(1000), np.random.random(1000))
+            ]
+        )
+
+        df1 = GeoDataFrame(
+            {"val1": np.random.randn(len(triangles)), "geometry": triangles}
+        )
+        df2 = GeoDataFrame({"val1": np.random.randn(len(points)), "geometry": points})
+
+        # save dfs
+        self.df, self.df2, self.data = df1, df2, df2.geometry.values.data
+        # cache bounds so that bound creation is not counted in benchmarks
+        self.bounds = [point.bounds for point in points]
+
+    def time_sindex_index_creation(self, *args):
+        self.df._invalidate_sindex()
+        self.df._generate_sindex()
```

Comment:
This benchmark is not necessarily useful for pygeos, because it only creates the actual tree on first query (at least not to draw any conclusion from the timing comparison with rtree, we can still use it to check there are no regressions in that part of course)

Reply:
Interesting! I'll add a comment in the benchmark so at least it's not confusing in the future.